modules/linkPreview/logic/linkPreviewLogic.js

```diff
@@ -17,40 +17,59 @@ const { LinkPreview } = require('../../../models');
 const DomLoader = require('../utils/domLoader');
 const cache = require('../../cache/utils/cache');
 const imageLogic = require('../../media/logic/imageLogic');
+const { MESSAGE_QUEUES, MESSAGES } = require('../../queue/constants/queue');
+const queue = require('../../queue/logic/queueLogic');
+const CacheLogic = require('../../cache/logic/cacheLogic');
+
 const logger = require('../../../utils/logger')(module);
 
 lngDetector.setLanguageType('iso2');
 
-module.exports = class LinkPreviewLogic {
-  static async fetchAllLinkPreviews() {
+class LinkPreviewLogic {
+  constructor() {
+    queue.subscribeToMessage(MESSAGE_QUEUES.LINKPREVIEW, MESSAGES.LINKPREVIEW.COMMANDS.UPDATE_DB,
+      async message => {
+        await this.updateLinkpreviewDatabase();
+        await queue.deleteMessage(MESSAGE_QUEUES.LINKPREVIEW, message.id);
+      });
```

Comment:
> Also, I expected that call will be more specific like to generate a preview for a specific tip invoked by tip logic/blockchain watcher. 

Thats the point we are working towards. This setup provides a first implementation of the message queue. 

>In comparison right now we don't need to remember all updateLinkpreviewDatabase calls just the last one because they do the same that makes the queue an overkill for now.

In this case it is overkill, I agree but its mostly for us to get some messages in the system and test its stability and add monitoring possibilities.

> Internally, queue can remove the message if the handler succeeds and to call the handler for all unhandled invocations after registration.

I am not sure, I think for now the handler can take care of the message, although its not ideal as it leads to code duplication I think the explicit action here requires the dev to think about how failures etc should be handled.